embedded-graphics/src/fonts/text.rs

```diff
@@ -0,0 +1,285 @@
+use crate::geometry::{Point, Size};
+use crate::pixelcolor::PixelColor;
+use crate::style::{Styled, TextStyle};
+use crate::transform::Transform;
+use crate::{Dimensions, DrawTarget};
+
+/// A text object.
+///
+/// The `Text` struct represents a string that can be drawn onto a display.
+///
+/// The text object only contains the string and position and no additional styling information,
+/// like the font or color. To draw a text object it is necessary to attach a style to it by using
+/// the [`into_styled`] method to create a [`Styled`] object.
+///
+/// See the [module-level documentation] for examples how to use text objects.
+///
+/// [`into_styled`]: #method.into_styled
+/// [`Styled`]: ../style/struct.Styled.html
+/// [module-level documentation]: index.html
+#[derive(Debug, PartialEq, Eq)]
+pub struct Text<'a> {
+    /// The string.
+    pub text: &'a str,
+
+    /// The position.
+    ///
+    /// The position of a text is defined as the top left pixel of the text.
+    pub position: Point,
+}
+
+impl<'a> Text<'a> {
+    /// Creates a text.
+    pub const fn new(text: &'a str, position: Point) -> Self {
+        Self { text, position }
+    }
+
+    /// Attaches a text style the the text object.
+    pub fn into_styled<C, F>(self, style: TextStyle<C, F>) -> Styled<Self, TextStyle<C, F>>
+    where
+        C: PixelColor,
+        F: Font,
+    {
+        Styled::new(self, style)
+    }
+}
+
+impl Transform for Text<'_> {
+    fn translate(&self, by: Point) -> Self {
+        Self {
+            position: self.position + by,
+            ..*self
+        }
+    }
+
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.position += by;
+
+        self
+    }
+}
+
+impl<C, F> Drawable<C> for &Styled<Text<'_>, TextStyle<C, F>>
+where
+    C: PixelColor,
+    F: Font + Copy,
+{
+    fn draw<T: DrawTarget<C>>(self, display: &mut T) {
+        display.draw_iter(self.into_iter());
+    }
+}
+
+impl<'a, C, F> IntoIterator for &Styled<Text<'a>, TextStyle<C, F>>
+where
+    C: PixelColor,
+    F: Font + Copy,
+{
+    type Item = Pixel<C>;
+    type IntoIter = StyledTextIterator<'a, C, F>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        Self::IntoIter {
+            current_char: self.primitive.text.chars().next(),
+            idx: 0,
+            text: self.primitive.text,
+            char_walk_x: 0,
+            char_walk_y: 0,
+            pos: self.primitive.position,
+            style: self.style,
+        }
+    }
+}
+
+impl<C, F> Dimensions for Styled<Text<'_>, TextStyle<C, F>>
+where
+    C: PixelColor,
+    F: Font,
+{
+    fn top_left(&self) -> Point {
+        self.primitive.position
+    }
+
+    fn bottom_right(&self) -> Point {
+        self.top_left() + self.size()
+    }
+
+    /// Returns the size of the bounding box of a styled text.
+    ///
+    /// Currently does not handle newlines (but neither does the rasteriser).
+    /// It will return `Size::zero()` if the string to render is empty.
```

Comment:
```suggestion
    /// It will return [`Size::zero()`] if the string to render is empty.
    ///
    /// [`Size::zero()`]: struct.Size.html#method.zero
```